experiments/snippets/smartAdopt/SmartAdopt.js

```diff
@@ -0,0 +1,193 @@
+var params = arguments[0],
+	sourceModelName = params.hasOwnProperty("$Input") && params.$Input.hasOwnProperty("SourceModel") ? params.$Input.SourceModel : false,
+	sourceModelIDField = params.hasOwnProperty("$Input") && params.$Input.hasOwnProperty("SourceModelIDField") ? params.$Input.SourceModelIDField : false,
+	sourceModel = sourceModelName ? skuid.$M(sourceModelName) : false,
+	sourceModelRows = sourceModel ? sourceModel.getRows() : false,
+	destModelName = params.hasOwnProperty("$Input") && params.$Input.hasOwnProperty("DestinationModel") ? params.$Input.DestinationModel : false,
+	destModelIDField = params.hasOwnProperty("$Input") && params.$Input.hasOwnProperty("DestinationModelIDField") ? params.$Input.DestinationModelIDField : false,
+	destModel = destModelName ? skuid.$M(destModelName) : false,
+	destModelRows = destModel ? destModel.getRows() : false,
+	create = params.hasOwnProperty("$Input") && params.$Input.hasOwnProperty("Create") ? trueish(params.$Input.Create) : false,
+	update = params.hasOwnProperty("$Input") && params.$Input.hasOwnProperty("Update") ? trueish(params.$Input.Update) : false,
+	del = params.hasOwnProperty("$Input") && params.$Input.hasOwnProperty("Delete") ? trueish(params.$Input.Delete) : false,
+	append = params.hasOwnProperty("$Input") && params.$Input.hasOwnProperty("Append") ? trueish(params.$Input.Append) : false,
+	idOnly = params.hasOwnProperty("$Input") && params.$Input.hasOwnProperty("IDOnly") ? trueish(params.$Input.IDOnly) : false,
+	excludes = params.hasOwnProperty("$Input") && params.$Input.hasOwnProperty("ExcludeFields") ? params.$Input.ExcludeFields : "",
+	includes = params.hasOwnProperty("$Input") && params.$Input.hasOwnProperty("IncludeFields") ? params.$Input.IncludeFields : "",
+	sourceRecordRegistry = {}, //Records with ID's
+	sourceRecordNew = [], //Records without ID's, presumably new records
+	destRecordRegistry = {}, //Records with ID's
+	contextRows = params.hasOwnProperty("context") && params.context.hasOwnProperty("rows") ?
+		params.context.rows :
+		params.hasOwnProperty("context") && params.context.hasOwnProperty("rows") ?
+			[params.context.row] :
+			false,
+	contextModel = contextRows && contextRows.length && contextRows[0].hasOwnProperty("model") ?
+		contextRows[0].model :
+		params.hasOwnProperty("context") && params.context.hasOwnProperty("model") ?
+			params.context.model :
+			false
+;
+
+excludes = excludes.split(",");
+includes = includes.split(",");
+for (var e = 0; e < excludes.length; e++) excludes[e] = excludes[e].trim();
+for (var f = 0; f < includes.length; f++) includes[f] = includes[f].trim();
+
+if (excludes.length === 1 && excludes[0] === "") excludes = [];
+if (includes.length === 1 && includes[0] === "") includes = [];
+
+var fieldMap = buildFieldMap(sourceModel, sourceModelIDField, destModel, destModelIDField, idOnly, excludes, includes);
+if (!fieldMap) return false;
```

Comment:
Build error here: `Parsing error: 'return' outside of function`

Reply:
I don't really know what to do with this. Skuid wraps snippets in a function, so this is valid (and necessary) for Skuid, but the build check on github can't really know this. If there was a way to disable this check for snippets (or all of skuid labs) that might help.

Reply:
#62 should resolve this issue, and thus no changes needed here. Resolving this comment.